### PR TITLE
Add E2E Test of the live preview feature by Playwright

### LINKED ISSUE #132: E2E Testing the live preview feature of charites by playwright
**Is your feature request related to a problem? Please describe.**
- The live preview feature is very important for charites.
- However, this functionality is written in Javascript, not Typescript, and is not software tested, so it is easily broken if changed.

**Describe the solution you'd like**
- So, I propose to add E2E testing by playwright for the live preview feature
- E2E testing can be very costly, but is worth considering for testing critical functions.

**Describe alternatives you've considered**
- Another option would be to Typescript-ize the Live Preview feature, but this would involve major changes.

**Additional context**
- I fixed the LegendControl feature in the live preview that was broken sometime ago
- My previous pull request broke the live preview feature.
  - #119
- It have fixed in this pull request.
  - #129
- As such, the live preview function is difficult to manually verify the operation for various cases and can be easily broken
- I want to prevent this from happening again.
- I have already started to implement Playwright with the following pull request
  - #131

### PR DESCRIPTION
**Pull request by yuiseki**
## Description

Resolve: #132 
~~Blocked by: #129~~

- Added minimal E2E testing by Playwright to prevent breaking the Live Preview feature
- At this point, I am only testing that there is no error output to the browser console at all
- In addition, I made sure to test for 3 patterns: maplibre, mapbox and geolonia
- And I also made this test run on GitHub Actions
  - Dependencies of Playwright is too big, so I decide to use `actions/cache` in GitHub Actions
    - Using cache make 10x faster when not using cache!
- ~~Right now, the live preview with mapbox is broken in the latest main branch, so the test is failing as intended!~~
  - Once the following pull request is merged, the test should pass
  - #129
- #129 has merged
  - So the test of this pull request sould be passed

### ~~IMPORTANT NOTE~~

- Testing live preview with mapbox on local dev env has required to set the environment variable named `MAPBOX_ACCESS_TOKEN`
- Also testing on the GitHub Actions has required to add the `Environment secrets` on the repository settings
- I am not authorized to manipulate the Secret of this repository, so I need someone who is authorized to do so.

## Type of Pull Request

- [x] Others (Adding more tests)

## Verify the followings

- [x] Code is up-to-date with the `main` branch
- [x] No build errors after `npm run build`
- [x] No lint errors after `npm run lint`
- [x] No errors on using `charites help` globally
- [x] Make sure all the exsiting features working well
- [x] Have you added at least one unit test if you are going to add new feature?
- [x] Have you updated documentation?

Refer to [CONTRIBUTING.MD](https://github.com/unvt/charites/tree/master/.github/CONTRIBUTING.md) for more details.

## Screenshots

[Image from Gyazo](https://gyazo.com/5e6fd0b86cf9165a0ee54eedc9195293)

### CODE REVIEW

**Comment by keichan34** [comment]
Thanks! This is a good idea. I'll try to find someone with access to the repository secrets (I don't either... maybe @miya0001 ?) I'll have to check about whose (what?) Mapbox account we should use to create the Mapbox key as well...

**Comment by keichan34** [comment]
@yuiseki I've added the mapbox token as `secrets.MAPBOX_ACCESS_TOKEN`

**Comment by yuiseki** [comment]
Currently I change the status of this pull request to draft because fix and check mapbox secret.
I will soon re-request review, please wait!

**Comment by smellman** [comment]
@yuiseki You need merge main branch.

**Comment by yuiseki** [comment]
- [x] I've merge current `main` branch into this branch
- [x] I've made changes to use `secrets.MAPBOX_ACCESS_TOKEN` . Thanks @keichan34 !
- [x] I've made change to use `actions/cache` in GitHub Action for speed up this test
- [x] Finally, The test that had been added by this pull request have passed
- [x] I've updated the description of this pull request

This pull request is ready for review 🙏

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -30,6 +28,33 @@ jobs:
...
+        id: playwright-cache
+        with:
+          path: |
+            ~/.cache/ms-playwright
+          key: ${{ runner.os }}-playwright-${{ hashFiles('**/package-lock.json') }}
+      - run: npm install
+      - run: npx playwright install --with-deps chromium
+        if: steps.playwright-cache.outputs.cache-hit != 'true'
+      - run: npm run build
+      - run: chmod 777 dist/cli.js
```
Comment (yuiseki) [inline .github/workflows/build.yml:53]: I'm wondering why I need to `chmod 777 dist/cli.js` before running `dist/cli.js`.
If I don't do this, I get a permission error.
It's might be a bug, but I believe that should be clearly describe and fix in another issue/pull request.

Comment (keichan34) [reply .github/workflows/build.yml:53]: TypeScript won't set execute permissions. I believe npm will set execute permissions to the files in the ["bin" section](https://github.com/unvt/charites/blob/main/package.json#L5-L7) when running npm package (or maybe npm install?). Here we can do `chmod +x dist/cli.js` or run it via node like `node dist/cli.js`. I don't think it's a big problem.

**Review by keichan34 — APPROVED** [review]
Looks good. Thanks!

**Review by naogify — APPROVED** [review]
@yuiseki 
Thank you! I confirmed that `npm run test:e2e` works correctly.

**Pull request merged**